hw/ip/otbn/dv/otbnsim/sim/insn.py

```diff
@@ -530,7 +530,7 @@ def execute(self, model: OTBNModel) -> None:
                              self.shift_bytes)
         (result,
          flags) = model.add_with_carry(a, -b_shifted,
-                                       1 - model.state.flags[self.flag_group].C)
+                                       -model.state.flags[self.flag_group].C)
```

Comment:
This does assume C is 0 or 1, we may want an assertion here to check this.

Reply:
I agree: please could you add the assertion? (Eventually, these flags should all be bools: I should probably add that to my ever growing todo list :-))